trpcultivate_germplasm/src/Plugin/TripalImporter/GermplasmCrossImporter.php

```diff
@@ -0,0 +1,972 @@
+use Drupal\trpcultivate\Plugin\Validators\GermplasmNameExists;
+use Drupal\trpcultivate\Plugin\Validators\ValidDelimitedFile;
+use Drupal\trpcultivate\Plugin\Validators\ValidHeaders;
+use Drupal\trpcultivate\Plugin\Validators\ValueInList;
+use Drupal\trpcultivate\TripalCultivateValidator\TripalCultivateValidatorManager;
+use Drupal\trpcultivate\Service\TripalCultivateFileTemplateService;
+use Drupal\trpcultivate\Service\ImportValidationHelper;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Tripal Cultivate Germplasm - Cross Importer.
+ *
+ * An importer for germplasm crosses developed in a breeding program.
+ *
+ * @TripalImporter(
+ *   id = "trpcultivate-germplasm-cross-importer",
+ *   label = @Translation("Tripal Cultivate: Germplasm Cross Importer"),
+ *   description = @Translation("Loads germplasm crosses into the system. This is useful for large datasets to ease the upload process."),
+ *   file_types = {"tsv"},
+ *   upload_description = @Translation("Please provide a data file."),
+ *   upload_title = @Translation("Germplasm Cross Data File*"),
+ *   use_analysis = FALSE,
+ *   require_analysis = FALSE,
+ *   use_button = True,
+ *   submit_disabled = FALSE,
+ *   button_text = "Import",
+ *   file_upload = TRUE,
+ *   file_local = FALSE,
+ *   file_remote = FALSE,
+ *   file_required = TRUE,
+ *   cardinality = 1,
+ *   menu_path = "",
+ *   callback = "",
+ *   callback_module = "",
+ *   callback_path = "",
+ * )
+ */
+#[TripalImporter(
+   id: 'trpcultivate-germplasm-cross-importer',
+   label: new TranslatableMarkup('Tripal Cultivate: Germplasm Cross Importer'),
+   description: new TranslatableMarkup('Loads germplasm crosses into the system. This is useful for large datasets to ease the upload process.'),
+   file_types: ['tsv'],
+   upload_description: new TranslatableMarkup('Please provide a data file.'),
+   upload_title: new TranslatableMarkup('Germplasm Cross Data File*'),
+   use_analysis: FALSE,
+   require_analysis: FALSE,
+   use_button: TRUE,
+   submit_disabled: FALSE,
+   button_text: new TranslatableMarkup('Import'),
+   file_upload: TRUE,
+   file_local: FALSE,
+   file_remote: FALSE,
+   file_required: TRUE,
+   cardinality: 1,
+   menu_path: '',
+   callback: '',
+   callback_path: '',
+  )]
+class GermplasmCrossImporter extends ChadoImporterBase implements ContainerFactoryPluginInterface {
+
+  use StringTranslationTrait;
+
+  /**
+   * Headers required by this importer.
+   *
+   * @var array
+   *
+   * The following keys are required:
+   * - 'name': The column header name as it should appear in the input file.
+   * - 'description': A user-friendly description of the header that will be
+   *   displayed to the user through the form.
+   * - 'type': one of "required" or "optional" to indicate whether the column
+   *   needs to have values present or not.
+   *
+   * NOTE: Order MUST reflect the desired order of headers in the input file.
+   */
+  private array $headers = [
```

Comment:
We want to be VERY selective in what we use private for. Any private variable will not be available to any importer extending this one... which I don't think is what we want for this variable or any of the others marked private in this class. Additionally these can cause issues with dependency injection serialization (hard to explain).